share/completions/solc.fish

```diff
@@ -0,0 +1,83 @@
+
+# Assembly Mode Options
+complete -c solc -l machine -d "Target machine in assembly or Yul mode."
+complete -c solc -l yul-dialect -d "Input dialect to use in assembly or yul mode."
+
+# Linker Mode Options
+complete -c solc -l libraries -d "Direct string or file containing library addresses."
+
+# Output Formatting
+complete -c solc -l pretty-json -d "Output JSON in pretty format."
+complete -c solc -l json-indent -d "Indent pretty-printed JSON with N spaces. Enables '--pretty-json' automatically."
+complete -c solc -l color -d "Force colored output."
+complete -c solc -l no-color -d "Explicitly disable colored output, disabling terminal auto-detection."
+complete -c solc -l error-codes -d "Output error codes."
+
+# Output Components
+complete -c solc -l ast-compact-json -d "AST of all source files in a compact JSON format."
+complete -c solc -l asm -d "EVM assembly of the contracts."
+complete -c solc -l asm-json -d "EVM assembly of the contracts in JSON format."
+complete -c solc -l opcodes -d "Opcodes of the contracts."
+complete -c solc -l bin -d "Binary of the contracts in hex."
+complete -c solc -l bin-runtime -d "Binary of the runtime part of the contracts in hex."
+complete -c solc -l abi -d "ABI specification of the contracts."
+complete -c solc -l ir -d "Intermediate Representation (IR) of all contracts (EXPERIMENTAL)."
+complete -c solc -l ir-optimized -d "Optimized intermediate Representation (IR) of all contracts (EXPERIMENTAL)."
+complete -c solc -l ewasm -d "Ewasm text representation of all contracts (EXPERIMENTAL)."
+complete -c solc -l hashes -d "Function signature hashes of the contracts."
+complete -c solc -l userdoc -d "Natspec user documentation of all contracts."
+complete -c solc -l devdoc -d "Natspec developer documentation of all contracts."
+complete -c solc -l metadata -d "Combined Metadata JSON whose Swarm hash is stored on-chain."
+complete -c solc -l storage-layout -d "Slots, offsets and types of the contract's state variables."
+
+# Extra Output:
+complete -c solc -l gas -d "Print an estimate of the maximal gas usage for each function."
+complete -c solc -l combined-json -d "Output a single json document containing the specified information."
+
+# Metadata Options:
+complete -c solc -l metadata-hash -d "Choose hash method for the bytecode metadata or disable it."
+complete -c solc -l metadata-literal -d " Store referenced sources as literal data in the metadata output."
+
+# Optimizer Options:
+complete -c solc -l optimize -d "Enable bytecode optimizer."
+complete -c solc -l optimize-runs -d "The number of runs specifies roughly how often each opcode of the deployed code will be executed across the lifetime of the contract. Lower values will optimize more for initial deployment cost, higher values will optimize more for high-frequency usage."
```

Comment:
Should probably add `-rf`, because it takes a non-file parameter. Same for others like model-checker-timeout